### ISSUE TITLE
App "blinky_freertos" cannot be built using CLANG

### ISSUE THREAD

**Issue by JuanSapriza** [esl-epfl/x-heep#263]
During the test for PR #239 it was seen that an issue was introduced into the `blinky_freertos` app, preventing it from being built. This does not happen when building with GCC.
This blocks the deployment of the CI of #239
Below the output of the compiler. This was built with Clang on the [X-HEEP builder docker container](https://github.com/orgs/esl-epfl/packages/container/package/xheep-compiler).

```
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /tmp/cc0akQCC.o: in function `vector_table':
/__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:22: undefined reference to `freertos_risc_v_trap_handler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /tmp/cc0akQCC.o: in function `.L0 ':
/__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:28: undefined reference to `freertos_risc_v_trap_handler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:36: undefined reference to `freertos_risc_v_mtimer_interrupt_handler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:44: undefined reference to `freertos_risc_v_trap_handler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:54: undefined reference to `freertos_risc_v_trap_handler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:56: undefined reference to `freertos_risc_v_trap_handler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:58: undefined reference to `freertos_risc_v_trap_handler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:60: undefined reference to `freertos_risc_v_trap_handler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /tmp/cc0akQCC.o:/__w/x-heep/x-heep/sw/device/lib/crt_freertos/vectors.S:62: more undefined references to `freertos_risc_v_trap_handler' follow
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/build/CMakeFiles/main.elf.dir/applications/blinky_freertos/main.c.obj: in function `no symbol':
main.c:(.text+0x1c2): undefined reference to `xTaskIncrementTick'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: main.c:(.text+0x1cc): undefined reference to `vTaskSwitchContext'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: main.c:(.text+0x1e2): undefined reference to `xTaskGetTickCountFromISR'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: main.c:(.text+0x24a): undefined reference to `xQueueGenericCreate'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/build/CMakeFiles/main.elf.dir/applications/blinky_freertos/main.c.obj: in function `.LBB2_5':
main.c:(.text+0x278): undefined reference to `xTaskCreate'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/build/CMakeFiles/main.elf.dir/applications/blinky_freertos/main.c.obj: in function `.LBB2_7':
main.c:(.text+0x29a): undefined reference to `xTaskCreate'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/build/CMakeFiles/main.elf.dir/applications/blinky_freertos/main.c.obj: in function `no symbol':
main.c:(.text+0x2b4): undefined reference to `vTaskStartScheduler'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: main.c:(.text+0x30c): undefined reference to `xQueueReceive'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: main.c:(.text+0x34c): undefined reference to `xTaskGetTickCount'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: main.c:(.text+0x376): undefined reference to `xTaskDelayUntil'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: main.c:(.text+0x386): undefined reference to `xQueueGenericSend'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/build/CMakeFiles/main.elf.dir/applications/blinky_freertos/main.c.obj: in function `.LBB6_3':
main.c:(.text+0x470): undefined reference to `xCriticalNesting'
/home/root/tools/riscv/lib/gcc/riscv32-unknown-elf/11.1.0/../../../../riscv32-unknown-elf/bin/ld: /__w/x-heep/x-heep/sw/build/CMakeFiles/main.elf.dir/applications/blinky_freertos/main.c.obj: in function `.LBB8_3':
main.c:(.text+0x4cc): undefined reference to `xCriticalNesting'
collect2: error: ld returned 1 exit status
```

**Comment by JuanSapriza** [comment]
When the CMakeLists.txt file was modified in #258, a part was removed that added the FreeRTOS files for the clang compiler. 
See highlighted above. 

![image](https://github.com/esl-epfl/x-heep/assets/54960111/f9517eea-f06a-4f93-8de3-a415944a669e)

@JoseCalero Was this done on purpose? 

Leaving the second option from the above image allows this app to be built using clang.

**Comment by JoseCalero** [comment]
fixed by #264
